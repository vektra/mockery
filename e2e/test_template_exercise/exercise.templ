.PkgName: {{ .PkgName }}
.SrcPkgQualifier: {{ .SrcPkgQualifier }}
{{ range $i, $import := .Imports }}
$import.Alias: {{ $import.Alias }}
$import.Path: {{ $import.Path }}
$import.Qualifier: {{ $import.Qualifier }}
{{ end }}

{{ range $i, $mock := .Mocks }}
# MOCK: {{ $i }}
$mock.InterfaceName: {{ $mock.InterfaceName }}
$mock.MockName: {{ $mock.MockName }}

{{ range $j, $typeparam := .TypeParams }}
# TYPE PARAM: {{ $j }}
$typeparam.Var.Name: {{ $typeparam.Var.Name }}
$typeparam.Var.IsSlice: {{ $typeparam.Var.IsSlice }}
$typeparam.Var.Nillable: {{ $typeparam.Var.Nillable }}
$typeparam.Var.Type.String: {{ $typeparam.Var.Type.String }}
$typeparam.Var.TypeString: {{ $typeparam.Var.TypeString }}
$typeparam.CallName: {{ $typeparam.CallName true }}
$typeparam.CallName: {{ $typeparam.CallName false }}
$typeparam.MethodArg: {{ $typeparam.MethodArg }}
$typeparam.Name: {{ $typeparam.Name }}
$typeparam.TypeString: {{ $typeparam.TypeString }}
$typeparam.TypeStringEllipsis: {{ $typeparam.TypeStringEllipsis }}
$typeparam.TypeStringVariadicUnderlying: {{ $typeparam.TypeStringVariadicUnderlying }}
{{- if $typeparam.Constraint }}
$typeparam.Constraint.String: {{ $typeparam.Constraint.String }}
{{- end }}
{{ end }}
{{ range $j, $method := .Methods }}
# METHOD: {{ $j }}
$method.Name: {{ $method.Name }}
$method.ReturnStatement: {{ $method.ReturnStatement }}
$method.Call: {{ $method.Call }}
$method.AcceptsContext: {{ $method.AcceptsContext}}
$method.Signature: {{ $method.Signature }}
$method.Declaration: {{ $method.Declaration }}
$method.ReturnsError: {{ $method.ReturnsError }}
$method.HasParams: {{ $method.HasParams }}
$method.HasReturns: {{ $method.HasReturns }}
{{ range $k, $param := .Params }}
# PARAM: {{ $k }}
$param.Var.Name: {{ $param.Var.Name }}
$param.Var.IsSlice: {{ $param.Var.IsSlice }}
$param.Var.Nillable: {{ $param.Var.Nillable }}
$param.Var.Type.String: {{ $param.Var.Type.String }}
$param.Var.TypeString: {{ $param.Var.TypeString }}
{{ end }}
{{ range $k, $return := .Returns }}
# RETURN: {{ $k }}
$return.Var.Name: {{ $return.Var.Name }}
$return.Var.IsSlice: {{ $return.Var.IsSlice }}
$return.Var.Nillable: {{ $return.Var.Nillable }}
$return.Var.Type.String: {{ $return.Var.Type.String }}
$return.Var.TypeString: {{ $return.Var.TypeString }}
{{ end }}
# SCOPE
$method.Scope.AllocateName "foo": {{ $method.Scope.AllocateName "foo" }}
$method.Scope.AllocateName "foo": {{ $method.Scope.AllocateName "foo" }}
$method.Scope.NameExists "foo": {{ $method.Scope.NameExists "foo" }}
$method.Scope.NameExists "foo1": {{ $method.Scope.NameExists "foo1" }}
$method.Scope.NameExists "notexists": {{ $method.Scope.NameExists "notexists" }}
{{ end }}
{{ end }}
