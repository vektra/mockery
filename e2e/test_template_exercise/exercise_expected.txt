.PkgName: test_pkgname
.SrcPkgQualifier: templateexercise.

$import.Alias: 
$import.Path: context
$import.Qualifier: context

$import.Alias: 
$import.Path: golang.org/x/exp/constraints
$import.Qualifier: constraints



# MOCK: 0
$mock.InterfaceName: Exercise
$mock.MockName: MockExercise


# TYPE PARAM: 0
$typeparam.Var.Name: T
$typeparam.Var.IsSlice: false
$typeparam.Var.Nillable: true
$typeparam.Var.Type.String: any
$typeparam.Var.TypeString: any
$typeparam.CallName: T
$typeparam.CallName: T
$typeparam.MethodArg: T any
$typeparam.Name: T
$typeparam.TypeString: any
$typeparam.TypeStringEllipsis: any
$typeparam.TypeStringVariadicUnderlying: any

# TYPE PARAM: 1
$typeparam.Var.Name: Ordered
$typeparam.Var.IsSlice: false
$typeparam.Var.Nillable: true
$typeparam.Var.Type.String: golang.org/x/exp/constraints.Ordered
$typeparam.Var.TypeString: constraints.Ordered
$typeparam.CallName: Ordered
$typeparam.CallName: Ordered
$typeparam.MethodArg: Ordered constraints.Ordered
$typeparam.Name: Ordered
$typeparam.TypeString: constraints.Ordered
$typeparam.TypeStringEllipsis: constraints.Ordered
$typeparam.TypeStringVariadicUnderlying: constraints.Ordered
$typeparam.Constraint.String: golang.org/x/exp/constraints.Integer


# METHOD: 0
$method.Name: Foo
$method.ReturnStatement: return
$method.Call: Foo(ctx, typeParam, ordered)
$method.AcceptsContext: true
$method.Signature: (ctx context.Context, typeParam T, ordered Ordered) (err error)
$method.Declaration: Foo(ctx context.Context, typeParam T, ordered Ordered) (err error)
$method.ReturnsError: true
$method.HasParams: true
$method.HasReturns: true

# PARAM: 0
$param.Var.Name: ctx
$param.Var.IsSlice: false
$param.Var.Nillable: true
$param.Var.Type.String: context.Context
$param.Var.TypeString: context.Context

# PARAM: 1
$param.Var.Name: typeParam
$param.Var.IsSlice: false
$param.Var.Nillable: true
$param.Var.Type.String: T
$param.Var.TypeString: T

# PARAM: 2
$param.Var.Name: ordered
$param.Var.IsSlice: false
$param.Var.Nillable: true
$param.Var.Type.String: Ordered
$param.Var.TypeString: Ordered


# RETURN: 0
$return.Var.Name: err
$return.Var.IsSlice: false
$return.Var.Nillable: true
$return.Var.Type.String: error
$return.Var.TypeString: error

# SCOPE
$method.Scope.AllocateName "foo": foo
$method.Scope.AllocateName "foo": foo1
$method.Scope.NameExists "foo": true
$method.Scope.NameExists "foo1": true
$method.Scope.NameExists "notexists": false


